### PR TITLE
Remove not needed statements that are also stored in SqlStatementRepository

### PR DESCRIPTION
**Pull request by sippsolutions**
Remove not needed statements that are also stored in SqlStatementRepository

### CODE REVIEW

**Comment by sippsolutions** [comment]
@wagnert

File: src/Utils/SqlStatementKeys.php [hunk 1]
```diff
@@ -87,98 +87,4 @@ class SqlStatementKeys extends \TechDivision\Import\Product\Utils\SqlStatementKe
...
-                    entity_id = :entity_id,
-                    request_path = :request_path,
-                    target_path = :target_path,
-                    redirect_type = :redirect_type,
-                    store_id = :store_id,
-                    description = :description,
-                    is_autogenerated = :is_autogenerated,
-                    metadata = :metadata
-              WHERE url_rewrite_id = :url_rewrite_id',
-        SqlStatementKeys::CREATE_URL_REWRITE_PRODUCT_CATEGORY =>
```
Comment (wagnert) [inline src/Utils/SqlStatementKeys.php:130]: This is not available in 'TechDivision\Import\Utils\SqlStatements' and can therefor not been deleted.

File: src/Utils/SqlStatementKeys.php [hunk 2]
```diff
@@ -87,98 +87,4 @@ class SqlStatementKeys extends \TechDivision\Import\Product\Utils\SqlStatementKe
...
-            'INSERT
-               INTO catalog_url_rewrite_product_category
-                    (url_rewrite_id,
-                     category_id,
-                     product_id)
-             VALUES (:url_rewrite_id,
-                     :category_id,
-                     :product_id)',
-        SqlStatementKeys::UPDATE_URL_REWRITE_PRODUCT_CATEGORY =>
```
Comment (wagnert) [inline src/Utils/SqlStatementKeys.php:139]: This is not available in 'TechDivision\Import\Utils\SqlStatements' and can therefor not been deleted.

File: src/Utils/SqlStatementKeys.php [hunk 3]
```diff
@@ -87,98 +87,4 @@ class SqlStatementKeys extends \TechDivision\Import\Product\Utils\SqlStatementKe
...
-            'UPDATE catalog_url_rewrite_product_category
-                SET category_id = :category_id,
-                    product_id = :product_id
-              WHERE url_rewrite_id = :url_rewrite_id',
-        SqlStatementKeys::DELETE_URL_REWRITE_PRODUCT_CATEGORY =>
```
Comment (wagnert) [inline src/Utils/SqlStatementKeys.php:144]: This is not available in 'TechDivision\Import\Utils\SqlStatements' and can therefor not been deleted.

File: src/Utils/SqlStatementKeys.php [hunk 4]
```diff
@@ -87,98 +87,4 @@ class SqlStatementKeys extends \TechDivision\Import\Product\Utils\SqlStatementKe
...
-            'DELETE
-               FROM catalog_url_rewrite_product_category
-              WHERE url_rewrite_id = :url_rewrite_id',
-        SqlStatementKeys::URL_REWRITE_PRODUCT_CATEGORY =>
```
Comment (wagnert) [inline src/Utils/SqlStatementKeys.php:148]: This is not available in 'TechDivision\Import\Utils\SqlStatements' and can therefor not been deleted.

File: src/Utils/SqlStatementKeys.php [hunk 5]
```diff
@@ -87,98 +87,4 @@ class SqlStatementKeys extends \TechDivision\Import\Product\Utils\SqlStatementKe
...
-            'SELECT *
-               FROM catalog_url_rewrite_product_category
-              WHERE url_rewrite_id = :url_rewrite_id',
-        SqlStatementKeys::URL_REWRITES_BY_SKU =>
```
Comment (wagnert) [inline src/Utils/SqlStatementKeys.php:152]: This is not available in 'TechDivision\Import\Utils\SqlStatements' and can therefor not been deleted.

File: src/Utils/SqlStatementKeys.php [hunk 6]
```diff
@@ -87,98 +87,4 @@ class SqlStatementKeys extends \TechDivision\Import\Product\Utils\SqlStatementKe
...
-            'SELECT t2.*
-               FROM catalog_product_entity t1,
-                    url_rewrite t2
-              WHERE t1.sku = :sku
-                AND t2.entity_id = t1.entity_id
-                AND t2.entity_type = \'product\'',
-        SqlStatementKeys::URL_REWRITE_PRODUCT_CATEGORIES_BY_SKU =>
```
Comment (wagnert) [inline src/Utils/SqlStatementKeys.php:159]: This is not available in 'TechDivision\Import\Utils\SqlStatements' and can therefor not been deleted.

**Review by wagnert — CHANGES_REQUESTED** [review]
Beside the comments in the code, it would be nice, if you can also delete the appropriate constants in the class, if not needed anymore.

**Comment by sippsolutions** [comment]
Hi @wagnert,

the deleted code seems to be copied from the SqlStatementRepository class.
The constructor tries to assign data to a variable called "preparedStatements" that's not available in SqlStatementKeys classes, but is available in Repositories.
Are you sure the mentioned keys are still needed?
If they are needed, a $preparedStatements private variable should be declared, otherwise the code should be removed (and the constants you mentioned aswell).

**Comment by wagnert** [comment]
Will be fixed with the next release.

**Pull request closed without merging**